src/features/cart/components/CartItem/CartItem.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { FC, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { toast } from "react-toastify";
+
+import { AppDispatch, RootState } from "@Store/store";
+import {
+  removeSingleCartItem,
+  changeCartItemCount,
+  mergeCartItemCategories,
+  ICartState,
+  selectCart,
+} from "@Cart/cartSlice";
+import { getNewCountAmount } from "@Cart/utils/cart";
+import { ERoutes } from "@/types/routes";
+import { ECount } from "@Products/types/product";
+import { ICartItemWithProduct } from "@Cart/types/cart";
+import Stars from "@CommonComponents/Stars/Stars";
+import Count from "@CommonComponents/Count/Count";
+import Modal from "@CommonComponents/Modal/Modal";
+
+import heart from "@Images/heart_thin.svg";
+import close from "@Images/close.svg";
+
+import "./CartItem.scss";
+
+interface CartItemProps {
+  itemWithProduct: ICartItemWithProduct;
+  invalidCategories: string[];
+}
+
+const CartItem: FC<CartItemProps> = ({
+  itemWithProduct,
+  invalidCategories,
+}) => {
+  const { product, count } = itemWithProduct;
+
+  const dispatch = useDispatch<AppDispatch>();
+  const navigate = useNavigate();
+  const { cart } = useSelector<RootState, ICartState>(selectCart);
+
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [modalText, setModalText] = useState<string>("");
+  const [nextCategory, setNextCategory] = useState<string>("");
+
+  const isCountInvalid =
+    count.amount > product.stock[count.category] ||
+    count.amount < ECount.MIN_COUNT_VALUE;
+  const priceSummary = (product.price[count.category] * count.amount).toFixed(
+    2
+  );
+
+  const notifyChangeCountCategory = (amount: number, category: string) =>
+    toast.warn(
+      `Amount was decreased! We have only ${amount} "${category}" in stock.`
+    );
+
+  const handleOpenModal = () => {
+    setShowModal(true);
+  };
+
+  const handleCloseModal = () => {
+    setShowModal(false);
+  };
+
+  const handleRemoveCartItem = () => {
+    dispatch(
+      removeSingleCartItem({ productId: product.id, category: count.category })
+    );
+  };
+
+  const handleOpenProduct = () => {
+    navigate(`/${ERoutes.PRODUCTS_LIST}/${product.id}`);
+  };
+
+  const handleChangeCategory = (newCategory: string) => {
+    setNextCategory(newCategory);
+    if (invalidCategories.includes(newCategory)) {
+      const nextAmount = getNewCountAmount(cart, product.id, newCategory);
+      let amountSum = nextAmount + count.amount;
+
+      if (amountSum > product.stock[newCategory]) {
+        amountSum = product.stock[newCategory];
```

Comment:
To avoid duplicates of product.stock[newCategory] you could create a variable

Reply:
Fixed - create a variable